include/dlaf/sender/transform_mpi.h

```diff
@@ -73,20 +84,20 @@ struct MPICallHelper {
     // whereas we are only looking to guard the submission of the MPI operation.
     // Moreover, the callable requires a Communicator to make its job, and it should be agnostic of
     // it being wrapped in a PromiseGuard or not.
-    // We therefore use unwrapPromiseGuardCommunicator to pass the Communicator& into a transformMPI
+    // We therefore use unwrapPromiseGuard to pass the Communicator& into a transformMPI
     // callables, then after returning from the callable, the PromiseGuard<Communicator> gets explicitly
-    // released using the helper consumePromiseGuard.
+    // released using the helper consumePromiseGuardCommunicator.
```

Comment:
If I get it correctly, this documentation has to be slightly revised. We have a "pre" and "post" phase.

In the "pre" phase, any `PromiseGuard`, no matter what type they are guarding, gets unwrapped and passed by reference to the function. Lifetime is kept by the `transformMPI`.

Inside `transformMPI`, after calling the function but before yielding for the result, there is the "post" phase.
Here ONLY `PromiseGuard<Communicator>` gets consumed manually in order to release it, because it is not needed anymore to keep the lifetime of the `Communicator` (required for posting, not for completing). But, other `PromiseGuard` have to be kept alive till the end of the function, so they don't get consumed.

Reply:
I just revised the doc of what happens for Communicator PromiseGuard.
It can be extended, however as #697 is modifying it again, I would leave the doc udate for that PR.